src/components/AutoSearch.tsx

```diff
@@ -73,7 +74,12 @@ export const AutoSearch: FC<AutoSearchProps> = ({
         setOpen(true);
       }}
       onChange={(_: object, value: string) => {
-        setValueFunction(value);
+        if (value) {
+          setValueFunction(value);
+          if (clearOnSelect) {
+            setDisplayValue(emptyValue);
+          }
+        }
```

Comment:
i references (der det kun er en rad med resultat), er vi avhengig av at setValueFunction blir kalt selv når value er null eller undefined, så fint om du kan endre dette til:
```
        setValueFunction(value ?? '');
        if (clearOnSelect) {
          setDisplayValue(emptyValue);
        }
```

Reply:
funksjonaliteten for references brekker hvis det står sånn som det er nå